compiler/src/dotty/tools/dotc/util/NameTransformer.scala

```diff
@@ -41,6 +41,18 @@ object NameTransformer {
   enterOp('?', "$qmark")
   enterOp('@', "$at")
 
+
+  /* Encode characters that are illegal */
+  def encodeSting(value: String): String = {
```

Comment:
Is this necessary? Can you `encode(name: SimpleName)` instead?

Reply:
I think so, `SimpleName` covers mutch more informations (start, length, next, ..)